Sprint 7

79137203595 opened the pull request:
Написаны unit- и UI-тесты.
Произведён рефакторинг кода на архитектуру MVP

79137203595 committed:
Выполнил 5 спринт

79137203595 committed:
исправил замечания ревьюера

79137203595 committed:
Добавил функцию для создания алерта

79137203595 committed:
6 спринт готовый на проверку

79137203595 committed:
1 действие рефакторинга, Создали файл с MovieQuizPresenter;
Перенесли туда метод convert(model: QuizQuestion);
Скопировали в presenter свойства questionsAmount и currentQuestionIndex;
Добавили свойство presenter в MovieQuizViewControler;
Применили DRY для устранения дублирования данных.

79137203595 committed:
кнопки да и нет

79137203595 committed:
presenter.didRecieveNextQuestion(question: question)

79137203595 committed:
Перенесём showNextQuestionOrResults в Presenter и заменим обращения к presenter на self:

79137203595 committed:
presenter.correctAnswers - убрал дубли

79137203595 committed:
прошел тест + приложение не падает

So1ovyov commented on MovieQuiz/Models/BestGame.swift:
```diff
@@ -0,0 +1,26 @@
+//
+
+import Foundation
+
```
Лишняя пустая строка, можно удалить.

So1ovyov commented on MovieQuiz/Models/MostPopularMovies.swift:
```diff
@@ -0,0 +1,38 @@
+            }
+            
+            return newURL
+        }
```
Сместились отступы, нужно поправить.

So1ovyov commented on MovieQuiz/Models/QuizQuestion.swift:
```diff
@@ -0,0 +1,40 @@
+
+import Foundation
+import UIKit
+//struct QuizQuestion {
```
Закомментированный код можно удалить, если он не используется.

So1ovyov commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,159 @@
+            self?.viewController?.show(quiz: viewModel)
+        }
+    }
+    func didLoadDataFromServer() {
```
Давай разделять методы 1-ой пустой строкой. чтобы легче читалось

So1ovyov commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,159 @@
+            questionFactory?.requestNextQuestion()
+        }
+    }
+    //    private func proceedToNextQuestionOrResults() {
```
Также закомментированный код

So1ovyov commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,159 @@
+            questionNumber: "\(currentQuestionIndex + 1)/\(questionsCount)")
+    }
+    
+    func yesButtonClicked() {
```
Давай этот метод и метод ниже объединим в один:
```
    func didTapButton(answer isCorrect: Bool) {
        didAnswer(isYes: isCorrect)
    }
```
Универсальный метод и кода меньше

So1ovyov commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,159 @@
+    
+    var questionsCount = 10
+    private var currentQuestionIndex = 0
+    var correctAnswers = 0
```
А точно ли эта переменная должна торчать наружу ? Кажется ее тоже можно сделать `private`

79137203595 committed:
Написаны и проходят unit-тесты:
- На сабскрипт для безопасного получения элемента массива по индексу (ArrayTests);
- На сервис MoviesLoader (для проверки успешной загрузки и для проверки ошибки MoviesLoaderTests).

79137203595 committed:
еще

79137203595 committed:
Написаны и проходят UI-тесты, исправил замечания проверки

79137203595 committed:
testYesButton и testNoButton проходят
